arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/IterableTest.kt

```diff
@@ -479,5 +479,25 @@ class IterableTest : UnitSpec() {
         listOfOption.flattenOption() shouldBe listOfOption.mapNotNull { it.orNull() }
       }
     }
+
+    "separateEither" {
+      checkAll(Arb.list(Arb.int())) { ints ->
+        val list = ints.map {
+          if (it % 2 == 0) it.left()
+          else it.right()
+        }
+        list.separateEither() shouldBe ints.partition { it % 2 == 0 }
```

Comment:
thanks @hoc081098 !. 

Maybe I'm missing something obvious or it's too late here but I believe this test is not correct because the assertion compares results of different types.

`Pair<List<Left<L>>, List<Right<R>>>` vs `Pair<List<Int>, List<Int>`. Just calling `partition` won't result in a new list of either but instead in a list of ints. 

What puzzles me is why tests are passing.

Reply:
`separateEither` returns `Pair<List<L>>, List<R>>` (values inside Lefts/Rights)

Reply:
My bad @hoc081098, for a second there I thought `separateEither` returned the results still in the Either structure.